examples/gpmdcov/gpmdcov.F90

```diff
@@ -1548,4 +1572,71 @@ subroutine gpmd_restart()
 
   end subroutine gpmd_restart
 
+  !> Reshuffle the parts
+  subroutine gpmd_reshuffle()
+
+    maxnparts = 0
+      do i=1,getNRanks()
+        np = gpat%localPartMax(i)-gpat%localPartMin(i)+1
+        maxnparts = max(maxnparts,np)
+      enddo
+
+
+    if(allocated(reshuffle))then
+      deallocate(reshuffle)
+      deallocate(PartsInRankI)
+      allocate(reshuffle(maxnparts,getNRanks()))
+      allocate(PartsInRankI(getNRanks()))
+    else
+      allocate(reshuffle(maxnparts,getNRanks()))
+      allocate(PartsInRankI(getNRanks()))
+    endif
+
+
+
+    reshuffle = 0
+    icount = 0
+    PartsInRankI = 0
+
+    do j=1,maxnparts
```

Comment:
The partitions are ordered in decreasing ch sizes (gpat%sgraph(ipt)%lsize). The reshuffling ensures that the load of the ranks get balanced.